third_party/move/move-compiler-v2/src/pipeline/control_flow_graph_simplifier.rs

```diff
@@ -0,0 +1,491 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Simplifies the control flow graph by
+//! (Implemented by `ControlFlowGraphSimplifier`)
```

Comment:
For each of these cases, please explicitly provide: (preconditions) before => after.

Reply:
done